MaxText/checkpointing.py

```diff
@@ -64,6 +72,53 @@ def create_orbax_checkpoint_manager(
   return mngr
 
 
+def create_orbax_emergency_checkpoint_manager(
+    local_checkpoint_dir: str,
+    persistent_checkpoint_dir: str,
+    global_mesh: jax.sharding.Mesh,
+    abstract_state: PyTree,
+    local_save_interval_steps: int,
+    persistent_save_interval_steps: int,
+):
+  """Returns an emergency checkpoint."""
+  max_logging.log("Creating emergency checkpoint manager...")
+
+  local_registry = type_handlers.create_type_handler_registry(
+      (
+          jax.Array,
+          type_handlers.ArrayHandler(primary_host=None, replica_id=None),
+      ),
+  )
+
+  local_checkpoint_handler = PyTreeCheckpointHandler(
+      use_ocdbt=True,
+      use_zarr3=True,
+      primary_host=None,
+      type_handler_registry=local_registry,
+  )
+
+  options = emergency_checkpoint_manager.CheckpointManagerOptions(
+      local=LocalCheckpointOptions(
+          save_interval_steps=local_save_interval_steps
+      ),
+      persistent=PersistentCheckpointOptions(
+          save_interval_steps=persistent_save_interval_steps
+      ),
+  )
+
+  emergency_mngr = emergency_checkpoint_manager.CheckpointManager(
+      local_checkpoint_dir,
+      epath.Path(persistent_checkpoint_dir),
+      global_mesh=global_mesh,
```

Comment:
Probably a good idea to include some assertions about the mesh configuration passed to CheckpointManager - make sure the 0th mesh_axes represents the slice.

Reply:
I would let that error out in Orbax, especially when we are considering supporting non-0th axis for `data`.